src/thor/bidirectional_astar.cc

```diff
@@ -137,134 +137,34 @@ void BidirectionalAStar::Init(const PointLL& origll, const PointLL& destll) {
-      }
-      // setup for expansion at this level
-      hierarchy_limits_forward_[node.level()].up_transition_count += trans->up();
-      const auto* trans_node = trans_tile->node(trans->endnode());
-      EdgeMetadata trans_meta =
-          EdgeMetadata::make(trans->endnode(), trans_node, trans_tile, edgestatus_forward_);
-      uint32_t trans_shortcuts = 0;
-      // expand the edges from this node at this level
-      for (uint32_t i = 0; i < trans_node->edge_count(); ++i, ++trans_meta) {
-        disable_uturn = ExpandForwardInner(graphreader, pred, trans_node, pred_idx, trans_meta,
-                                           trans_shortcuts, trans_tile, offset_time) ||
-                        disable_uturn;
-      }
-    }
-  }
-
-  // Now, after having looked at all the edges, including edges on other levels,
-  // we can say if this is a deadend or not, and if so, evaluate the uturn-edge (if it exists)
-  if (!disable_uturn && uturn_meta) {
-    // If we found no suitable edge to add, it means we're at a deadend
-    // so lets go back and re-evaluate a potential u-turn
-    pred.set_deadend(true);
-
-    // TODO Is there a shortcut that supersedes our u-turn?
-    // Decide if we should expand a shortcut or the non-shortcut edge...
-
-    // Expand the uturn possiblity
-    disable_uturn = ExpandForwardInner(graphreader, pred, nodeinfo, pred_idx, uturn_meta, shortcuts,
-                                       tile, offset_time) ||
-                    disable_uturn;
-  }
-
-  return disable_uturn;
-}
-
-// Runs in the inner loop of `ExpandForward`, essentially evaluating if
+// Runs in the inner loop of `Expand`, essentially evaluating if
 // the edge described in `meta` should be placed on the stack
 // as well as doing just that.
 //
-// TODO: Merge this with ExpandReverseInner
-//
 // Returns false if uturns are allowed.
 // Returns true if we will expand or have expanded from this edge. In that case we disallow uturns.
 // Some edges we won't expand from, but we will still put them on the adjacency list in order to
 // connect the forward and reverse paths. In that case we return false to allow uturns only if this
 // edge is a not-thru edge that will be pruned.
 //
-inline bool BidirectionalAStar::ExpandForwardInner(GraphReader& graphreader,
-                                                   const BDEdgeLabel& pred,
-                                                   const NodeInfo* nodeinfo,
-                                                   const uint32_t pred_idx,
-                                                   const EdgeMetadata& meta,
-                                                   uint32_t& shortcuts,
-                                                   const graph_tile_ptr& tile,
-                                                   const TimeInfo& time_info) {
+template <const BidirectionalAStar::ExpansionType expansion_direction>
+inline bool BidirectionalAStar::ExpandInner(baldr::GraphReader& graphreader,
```

Comment:
Curious, does `inline`ing here provide any speedup or is this required to get around ODR?

Reply:
@mandeepsandhu I don't think it actually provides any benefit - ODR is usually only a problem in header-defined templates, which this is not.

The real reason it's here is that it was here in the previous code, and I'm trying to minimize change.